File: services/api/src/kew_api/schemas/workspace.py
"""Workspace / documentation tree schemas."""

from __future__ import annotations

from typing import Literal

from pydantic import BaseModel, Field


class WorkspaceInfo(BaseModel):
    root: str
    name: str
    docs_root: str


class TreeNode(BaseModel):
    id: str
    name: str
    path: str
    type: Literal["folder", "file"]
    has_children: bool = False
    children: list[TreeNode] | None = None


class TreeResponse(BaseModel):
    parent_path: str
    nodes: list[TreeNode]


class CreateFolderRequest(BaseModel):
    parent_path: str = ""
    name: str = Field(min_length=1, max_length=255)


class CreateDocumentRequest(BaseModel):
    parent_path: str = ""
    name: str = Field(min_length=1, max_length=255)
    template_type: str = "overview"


class UpdateNodeRequest(BaseModel):
    path: str = Field(min_length=1)
    new_name: str | None = Field(default=None, min_length=1, max_length=255)
    new_parent_path: str | None = None


class DeleteNodeResponse(BaseModel):
    deleted: str
    type: Literal["folder", "file"]

File: apps/workbench/src/shared/types/workspace.ts
export type NodeType = "folder" | "file";

export interface TreeNode {
  id: string;
  name: string;
  path: string;
  type: NodeType;
  hasChildren: boolean;
  children?: TreeNode[] | null;
}

export interface TreeResponse {
  parentPath: string;
  nodes: TreeNode[];
}

export interface WorkspaceInfo {
  root: string;
  name: string;
  docsRoot: string;
}

export interface CreateFolderRequest {
  parentPath: string;
  name: string;
}

export interface CreateDocumentRequest {
  parentPath: string;
  name: string;
  templateType?: string;
}

export interface UpdateNodeRequest {
  path: string;
  newName?: string;
  newParentPath?: string;
}

export interface DeleteNodeResponse {
  deleted: string;
  type: NodeType;
}

export interface ApiErrorBody {
  code: string;
  message: string;
  details?: Record<string, unknown>;
}
